src/ccache.c

```diff
@@ -3360,12 +3383,18 @@ ccache(int argc, char *argv[])
 
 	cc_log("Object file: %s", output_obj);
 
+	if (conf->debug) {
+		debug_start(output_obj);
+	}
+
 	struct mdfour common_hash;
 	hash_start(&common_hash);
+	mdfour_identify(&common_hash, 'c');
 	calculate_common_hash(preprocessor_args, &common_hash);
 
 	// Try to find the hash using the manifest.
 	struct mdfour direct_hash = common_hash;
+	mdfour_identify(&direct_hash, 'd');
```

Comment:
As suggested in bullet 5 in #226, would it be possible to add a debug log file header here saying "here comes the direct mode part", and similar for the common and preprocessor parts?

Reply:
> for the binary version I wanted to keep the exact same format so that the checksums would match what is being used as the key in the cache.

Certainly, that's good.

> I did that (well, implicitly) for the text version,

I guess it's possible to deduce the three sections if you happen to know how ccache works in detail, but I think that it would be very nice with explicit section markers in the `*.ccache-input` file, perhaps like this:

```
=== COMMON ===
3
### ext
i
[...]

=== DIRECT MODE ===
### manifest version
1
### arg
-g
[...]

=== PREPROCESSOR MODE ===
### arg
-g
[...]
```

Reply:
Done